### ISSUE TITLE
Is there any chance to support libpolarssl / mbed TLS?

### ISSUE THREAD

**Issue by Arnie97** [aria2/aria2#741]
OpenSSL usually takes more space than aria2 itself, while mbed TLS (formerly PolarSSL) is targeted at embedded systems thus could provide a minimal footprint for embedded systems.

**Comment by nmaier** [comment]
We already support openssl, gnutls, Apple Secure Transport, Windows Schannel, so I see no reason why it cannot be supported too... It's just somebody has to actually write the code.

**Comment by tatsuhiro-t** [comment]
Adding another backend increases maintenance cost. Perhaps, we need to know how much mbed TLS cuts footprint against OpenSSL and GNUTLS builds to justify the benefits outweigh the added cost.

**Comment by nmaier** [comment]
Of course, when I said write the code, I really meant write *and maintain* the code.

**Comment by nmaier** [comment]
mbedTLS has financial backing and developer resources by ARM now, is highly configurable it seems to strip stuff you don't need/want and is designed to have a low memory (both binary and runtime) footprint they say.

A quick google turned up people creating mTLS libary configurations as small as 30KB and trying to get even smaller, another result showed people having a runtime (heap+stack, but not the image) of about 4K which includes any global/shared structures and a connection if I understood it correctly. A typical OpenSSL connection weights at about 128K it seems, while the binary size is in the megabytes.

Measuring what this means in practice is a chicken and egg problem tho: You cannot measure without having an implementation, and making the implementation dependent on what you measured... you see the problem.
However, I think one could do some similar measurements if one tests OpenSSL vs mTLS vs GnuTLS curl builds, as curl supports them all already, to get a general idea how much they differ.

I might actually look into it if I find some time, as I run aria2 on a low-powered, low-memory Atom SoC sometimes (not really comparable with even tinier ARMs, but still)
